### PR TITLE
Feature/mood meter

### PR DESCRIPTION
**Pull request by finncelta**
added the Mood Meter together with it's bootstrap and routing

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -2203,7 +2203,6 @@
       "version": "1.3.7",
       "resolved": "https://registry.npmjs.org/accepts/-/accepts-1.3.7.tgz",
```
Comment (waspmuc) [inline package-lock.json:2204]: @YeetGrandpa Kannst du mir sagen wozu/zu welchem Zweck dieses file (package-lock.json) da ist?

File: package.json [hunk 2]
```diff
@@ -22,7 +22,13 @@
     "@angular/platform-browser-dynamic": "~8.2.13",
     "@angular/router": "~8.2.13",
     "@ng-bootstrap/ng-bootstrap": "^5.1.4",
+    "angular-bootstrap-md": "^8.6.0",
+    "bootstrap": "^4.3.1",
+    "canvasjs": "^1.8.1",
+    "chart.js": "^2.9.3",
     "hammerjs": "^2.0.8",
+    "jquery": "^3.4.1",
```
Comment (waspmuc) [inline package.json:30]: @YeetGrandpa Kannst du mir sagen was ^ und ~ in den Versionsnummern bedeutet?

File: src/app/app.component.html [hunk 3]
```diff
@@ -1,2 +1,9 @@
-<app-nav-bar></app-nav-bar>
-<app-mood-ometer></app-mood-ometer>
+<nav class="navbar-nav fixed-top navbar-expand-md navbar-light bg-light Navigation">
+  <div id="logo"></div>
+  <nav style="padding-left: 10px; padding-top: 20px">
+    <a routerLink="/mood">MoodOMeter</a>
```
Comment (waspmuc) [inline src/app/app.component.html]: please use "moodometer", it's easier to follow what it is

File: src/app/app.component.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import { Component } from '@angular/core';
 
 
+
```
Comment (waspmuc) [inline src/app/app.component.ts]: Why did you add this empty line?

File: src/app/mood-ometer.service.ts [hunk 5]
```diff
@@ -1,4 +1,7 @@
 import { Injectable } from '@angular/core';
+import {HttpClient, HttpHeaders} from '@angular/common/http';
+import {BrowserModule} from '@angular/platform-browser';
+
```
Comment (waspmuc) [inline src/app/mood-ometer.service.ts:4]: remove additional empty line

File: src/app/mood-ometer/mood-ometer.component.html [hunk 6]
```diff
@@ -1,8 +1,18 @@
-
-<div class="content">
-<mat-radio-group class="radio-group" [(ngModel)]="userChoice">
-<mat-radio-button color="primary" class="radioButton" *ngFor="let possibility of possibilities" [value]="possibility">{{possibility}}</mat-radio-button>
-<button mat-raised-button class="button">Send</button>
-</mat-radio-group>
-
+<div class="container-fluid myContainer">
```
Comment (waspmuc) [inline src/app/mood-ometer/mood-ometer.component.html:1]: please find a proper name for that element

File: src/app/mood-ometer/mood-ometer.component.ts [hunk 7]
```diff
@@ -1,15 +1,26 @@
...
 @Component({
   selector: 'app-mood-ometer',
   templateUrl: './mood-ometer.component.html',
   styleUrls: ['./mood-ometer.component.css']
 })
 export class MoodOMeterComponent implements OnInit {
-userChoice: string;
+  constructor(private http: HttpClient) {}
+userChoice;
+url = 'http://10.3.4.114:4200/Gradle___Alor_Backend_war/rest/mdm/create';
```
Comment (waspmuc) [inline src/app/mood-ometer/mood-ometer.component.ts]: Use relative paths and have our discussion from yesterday in mind about different environments. 
--> https://stackoverflow.com/questions/48182290/how-to-inject-api-base-url-a-string-in-an-angular-service

File: src/app/mood-ometer/mood-ometer.component.ts [hunk 8]
```diff
@@ -1,15 +1,26 @@
...
+radioChoice: string;
```
Comment (waspmuc) [inline src/app/mood-ometer/mood-ometer.component.ts]: Find a proper name e.g. moodSelection or similar

File: src/app/mood-ometer/mood-ometer.component.ts [hunk 9]
```diff
@@ -1,15 +1,26 @@
...
 possibilities: string[] = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10'];
-
-  constructor() { }
+onItemChange(value) {
+  this.radioChoice = value;
+  console.log(' Wert: ', value);
+}
+  addEntry() {
```
Comment (waspmuc) [inline src/app/mood-ometer/mood-ometer.component.ts]: Find a proper name and use parametes in method call (even if you don't have to). It's all about reading the code easily and less error prone code.

File: src/app/pie-chart/pie-chart.component.ts [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  selector: 'app-pie-chart',
+  templateUrl: './pie-chart.component.html',
+  styleUrls: ['./pie-chart.component.css']
+})
+export class PieChartComponent {
+  constructor(private http: HttpClient) {
+  }
+
+  public chartType = 'pie';
+  url = 'http://10.3.4.114:4200/Gradle___Alor_Backend_war/rest/mdm/dist';
```
Comment (waspmuc) [inline src/app/pie-chart/pie-chart.component.ts]: same goes here: everytime you write the same code for the second time, find an abstraction for it.

Comment (waspmuc) [inline src/app/pie-chart/pie-chart.component.ts]: find a better name! url could be any url. Think about a webbapp with multiple API-Endpoints of multiple servers... how do you handle that? Url1, url2?...

File: src/app/userPick.ts [hunk 11]
```diff
@@ -0,0 +1 @@
+
```
Comment (waspmuc) [inline src/app/userPick.ts]: Why did you add the empty line here?

**Review by waspmuc — CHANGES_REQUESTED** [review]
Denkt bitte in Zukunft an all die Dinge, die ich euch bereits so oft gesagt habe. Es ist wichtiger, dass ich qualitativen Code schreibt als dass ihr viele Features fertig bekommt. Bitte weitersagen!

File: package-lock.json (hunk 1 above)
Comment (finncelta) [reply package-lock.json:2204]: Die wird von npm automatisch generiert, wenn npm bestimmte dateien modifiziert, damit bei späteren installs die richtigen unterordner verwendet werden.
https://docs.npmjs.com/files/package-lock.json

File: package.json (hunk 2 above)
Comment (finncelta) [reply package.json:30]: Das "^" bedeutet das auch neuere versionen verwendet werden dürfen und "~" beudetet das neuere versionen verwendet werden dürfen, wenn sie bugfixes enthalten.
https://bytearcher.com/articles/semver-explained-why-theres-a-caret-in-my-package-json/

File: src/app/app.component.html (hunk 3 above)
Comment (finncelta) [reply src/app/app.component.html]: will be fixed on next push

File: src/app/app.component.ts (hunk 4 above)
Comment (finncelta) [reply src/app/app.component.ts]: will be fixed on next push

File: src/app/mood-ometer.service.ts (hunk 5 above)
Comment (finncelta) [reply src/app/mood-ometer.service.ts:4]: will be fixed on next push

File: src/app/mood-ometer/mood-ometer.component.ts (hunk 8 above)
Comment (finncelta) [reply src/app/mood-ometer/mood-ometer.component.ts]: will be fixed on next push

File: src/app/userPick.ts (hunk 11 above)
Comment (finncelta) [reply src/app/userPick.ts]: file will be deleted on next push

File: src/app/pie-chart/pie-chart.component.ts (hunk 10 above)
Comment (finncelta) [reply src/app/pie-chart/pie-chart.component.ts]: url is renamed to urlMoodDistGet on next push

File: src/app/pie-chart/pie-chart.component.ts [hunk 12]
```diff
@@ -11,7 +11,7 @@ export class PieChartComponent {
   }
 
   public chartType = 'pie';
-  url = 'http://10.3.4.114:4200/Gradle___Alor_Backend_war/rest/mdm/dist';
+  urlMoodDistGet = 'http://10.3.4.114:4200/Gradle___Alor_Backend_war/rest/mdm/dist';
```
Comment (waspmuc) [inline src/app/pie-chart/pie-chart.component.ts:14]: urlMoodDistGet is still a meaningless name. If I don't have any knowledge about that code/project I can't make up anything out of that name. What about urlMoodDistribution?

**Review by waspmuc — CHANGES_REQUESTED** [review]
Small change left. Anything else is okay
